test/updater_unittest.cc

```diff
@@ -0,0 +1,78 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+#include <string>
+#include <vector>
+
+namespace google {
+
+class UpdaterTest : public ::testing::Test {
+ protected:
+  // query_metadata function not needed to test callbacks.
+  UpdaterTest() : config(), store(config), updater(
+      config, &store, "Test", 60.0, nullptr) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() const {
+    return store.GetMetadataMap();
+  }
+
+  bool ValidateConfiguration() const {
+    return updater.ValidateConfiguration();
+  }
+
+  void UpdateMetadataCallback(MetadataUpdater::ResourceMetadata&& result) {
+    updater.UpdateMetadataCallback(std::move(result));
+  }
+
+  void UpdateResourceCallback(const MetadataUpdater::ResourceMetadata& result) {
+    updater.UpdateResourceCallback(result);
+  }
+
+  Configuration config;
+  MetadataStore store;
+  PollingMetadataUpdater updater;
+};
+
+namespace {
+
+TEST_F(UpdaterTest, ValidateConfiguration) {
+  EXPECT_TRUE(ValidateConfiguration());
```

Comment:
Should we also check that `PollingMetadataUpdater::ValidateConfiguration` fails when the period is negative?
You would have to pull out the variables from the fixture and make the accessor function take an updater object instead, i.e.:
```cpp
  static bool ValidateConfiguration(const MetadataUpdater& updater) {
    return updater.ValidateConfiguration();
  }

  static void UpdateMetadataCallback(
      MetadataUpdater* updater, MetadataUpdater::ResourceMetadata&& result) {
    updater->UpdateMetadataCallback(std::move(result));
  }

  static void UpdateResourceCallback(
      MetadataUpdater* updater,
      const MetadataUpdater::ResourceMetadata& result) {
    updater->UpdateResourceCallback(result);
  }
```